glib/src/variant.rs

```diff
@@ -247,12 +247,26 @@ impl Variant {
     #[doc(alias = "get_child_value")]
     #[doc(alias = "g_variant_get_child_value")]
     pub fn child_value(&self, index: usize) -> Variant {
-        assert!(index < self.n_children());
         assert!(self.is_container());
+        assert!(index < self.n_children());
 
         unsafe { from_glib_full(ffi::g_variant_get_child_value(self.to_glib_none().0, index)) }
     }
 
+    /// Try to read a child item out of a container `Variant` instance.
+    ///
+    /// It returns `None` if `self` is not a container type or if the given
+    /// `index` is larger than number of children.
+    pub fn try_child_value(&self, index: usize) -> Option<Variant> {
```

Comment:
Do you see any use cases for this where the caller wouldn't immediately want to do `.get()` on this?  Searching the source...actually right now most of the uses of `child_value()` immediately call `.get()`.

Reply:
I think you are wondering about directly having a fallible generic getter. Yes, I think it would make sense, I also observed that pattern in basically all consumers. (Not sure if we are biased though as we look more or less at the same codebases, and I didn't experience many other glib-based projects other than those.)

Reply:
https://github.com/gtk-rs/gtk-rs-core/pull/226